File: PythonNCAA/scripts/_sweep_recent.mjs
#!/usr/bin/env node
import fs from "fs";

const store = JSON.parse(fs.readFileSync("data/history.json", "utf8"));
const runs = store.runs || [];

// Last 2 weeks = dates >= 20260303
const cutoff = "20260303";
const recentRuns = runs.filter(r => r.date >= cutoff);
const olderRuns = runs.filter(r => r.date < cutoff);

console.log(`Recent runs (>= ${cutoff}): ${recentRuns.length} days`);
console.log(`Older runs (< ${cutoff}): ${olderRuns.length} days\n`);

function gradeSpread(g, side) {
  const homeMargin = g.homeScore - g.awayScore;
  const v = homeMargin - g.line;
  if (side === "home") return v > 0 ? "WIN" : v === 0 ? "PUSH" : "LOSS";
  else return v < 0 ? "WIN" : v === 0 ? "PUSH" : "LOSS";
}

function sweep(runSet, label) {
  const games = [];
  for (const r of runSet) {
    for (const g of r.games) {
      if (g.homeScore === undefined || g.awayScore === undefined) continue;
      if (!g.line && g.line !== 0) continue;
      if (g.status === "SKIPPED" || g.status === "MISSING_STATS") continue;
      if (!g.pHomeCover && !g.pAwayCover) continue;
      games.push(g);
    }
  }

  console.log(`=== ${label} (${games.length} games) ===`);
  console.log("pCover | W    - L   - P | Pct%  | Picks/Day | Units");
  console.log("-------+----------------+-------+-----------+------");

  const days = runSet.length || 1;

  for (const pThresh of [0.55, 0.58, 0.60, 0.62, 0.65, 0.67]) {
    let w = 0, l = 0, p = 0;
    for (const g of games) {
      const pH = g.pHomeCover || 0;
      const pA = g.pAwayCover || 0;
      const bestP = Math.max(pH, pA);
      const side = pH >= pA ? "home" : "away";
      const absLine = Math.abs(g.line);

      // Dog check for fav line cap
      const isDog = side === "home" ? g.line < 0 : g.line > 0;
      const lineOK = isDog ? true : absLine <= 6;

      if (bestP >= pThresh && lineOK && absLine > 0) {
        const result = gradeSpread(g, side);
        if (result === "WIN") w++;
        else if (result === "LOSS") l++;
        else p++;
      }
    }
    const total = w + l;
    const pct = total > 0 ? (w / total * 100).toFixed(1) : "n/a";
    const units = (w * 1 - l * 1.1).toFixed(1);
    const ppd = ((w + l + p) / days).toFixed(1);
    console.log(`  ${pThresh.toFixed(2)}  | ${String(w).padStart(4)}-${String(l).padStart(4)}-${String(p).padStart(2)} | ${pct.padStart(5)} | ${ppd.padStart(9)} | ${units.padStart(6)}`);
  }
  console.log();
}

sweep(recentRuns, "LAST 2 WEEKS");
sweep(olderRuns, "BEFORE LAST 2 WEEKS");
sweep(runs, "FULL SEASON");

File: pyNCAA/scripts/_sweep_recent.py
#!/usr/bin/env python3
# scripts/_sweep_recent.py — Compare recent vs older performance
import json

store = json.loads(open("data/history.json").read())
runs = store.get("runs") or []

cutoff = "20260303"
recent_runs = [r for r in runs if r.get("date", "") >= cutoff]
older_runs = [r for r in runs if r.get("date", "") < cutoff]

print(f"Recent runs (>= {cutoff}): {len(recent_runs)} days")
print(f"Older runs (< {cutoff}): {len(older_runs)} days\n")

def grade_spread(g, side):
    margin = g["homeScore"] - g["awayScore"]
    v = margin - g["line"]
    if side == "home": return "WIN" if v > 0 else ("PUSH" if v == 0 else "LOSS")
    return "WIN" if v < 0 else ("PUSH" if v == 0 else "LOSS")

def sweep(run_set, label):
    games = []
    for r in run_set:
        for g in (r.get("games") or []):
            if g.get("homeScore") is None or g.get("awayScore") is None: continue
            if g.get("line") is None and g.get("line") != 0: continue
            if g.get("status") in ("SKIPPED", "MISSING_STATS"): continue
            if not g.get("pHomeCover") and not g.get("pAwayCover"): continue
            games.append(g)
    print(f"=== {label} ({len(games)} games) ===")
    print("pCover | W    - L   - P | Pct%  | Picks/Day | Units")
    days = len(run_set) or 1
    for p_thresh in [0.55, 0.58, 0.60, 0.62, 0.65, 0.67]:
        w, l, p = 0, 0, 0
        for g in games:
            pH, pA = g.get("pHomeCover", 0), g.get("pAwayCover", 0)
            best_p = max(pH, pA)
            side = "home" if pH >= pA else "away"
            abs_line = abs(g.get("line", 0))
            is_dog = (side == "home" and g["line"] < 0) or (side == "away" and g["line"] > 0)
            line_ok = True if is_dog else abs_line <= 6
            if best_p >= p_thresh and line_ok and abs_line > 0:
                result = grade_spread(g, side)
                if result == "WIN": w += 1
                elif result == "LOSS": l += 1
                else: p += 1
        total = w + l
        pct = f"{w / total * 100:.1f}" if total > 0 else "n/a"
        units = f"{(w - l * 1.1):.1f}"
        ppd = f"{(w + l + p) / days:.1f}"
        print(f"  {p_thresh:.2f}  | {w:>4}-{l:>4}-{p:>2} | {pct:>5} | {ppd:>9} | {units:>6}")
    print()

sweep(recent_runs, "LAST 2 WEEKS")
sweep(older_runs, "BEFORE LAST 2 WEEKS")
sweep(runs, "FULL SEASON")
